pkg/build/iso.go

```diff
@@ -1,39 +1,92 @@
 package build
 
 import (
+	_ "embed"
 	"fmt"
 	"os"
 	"os/exec"
 	"path/filepath"
-	"strings"
 
+	"github.com/suse-edge/edge-image-builder/pkg/fileio"
+	"github.com/suse-edge/edge-image-builder/pkg/template"
 	"go.uber.org/zap"
 )
 
 const (
-	xorrisoArgsBase = "-indev %s -outdev %s -map %s /combustion -boot_image any replay -changes_pending yes"
-	xorrisoExec     = "/usr/bin/xorriso"
-	xorrisoLogFile  = "iso-build.log"
+	isoExtractDir        = "iso-extract"
+	rawExtractDir        = "raw-extract"
+	extractIsoScriptName = "iso-extract.sh"
+	extractIsoLogFile    = "iso-extract.log"
+	rebuildIsoScriptName = "iso-build.sh"
+	rebuildIsoLogFile    = "iso-build.log"
 )
 
+//go:embed templates/extract-iso.sh.tpl
+var extractIsoTemplate string
+
+//go:embed templates/rebuild-iso.sh.tpl
+var rebuildIsoTemplate string
+
 func (b *Builder) buildIsoImage() error {
-	err := b.deleteExistingOutputIso()
-	if err != nil {
+	if err := b.deleteExistingOutputIso(); err != nil {
 		return fmt.Errorf("deleting existing ISO image: %w", err)
 	}
 
-	cmd, logfile, err := b.createXorrisoCommand()
+	if err := b.extractIso(); err != nil {
+		return fmt.Errorf("extracting the ISO image: %w", err)
+	}
+
+	// TODO: Call into raw code
+
+	if err := b.rebuildIso(); err != nil {
+		return fmt.Errorf("building the ISO image: %w", err)
+	}
+
+	return nil
+}
+
+func (b *Builder) extractIso() error {
+	if err := b.writeIsoScript(extractIsoTemplate, extractIsoScriptName); err != nil {
+		return fmt.Errorf("creating the ISO extraction script: %w", err)
+	}
+
+	cmd, extractLog, err := b.createIsoCommand(extractIsoLogFile, extractIsoScriptName)
 	if err != nil {
-		return fmt.Errorf("configuring the ISO build command: %w", err)
+		return fmt.Errorf("preparing to extract the contents of the ISO: %w", err)
+	}
+	defer func() {
+		if err = extractLog.Close(); err != nil {
+			zap.L().Warn("failed to close ISO extraction log file properly", zap.Error(err))
+		}
+	}()
+
+	if err = cmd.Run(); err != nil {
+		return fmt.Errorf("extracting the contents of the ISO: %w", err)
 	}
-	defer logfile.Close()
 
-	err = cmd.Run()
+	return nil
+}
+
+func (b *Builder) rebuildIso() error {
+	if err := b.writeIsoScript(rebuildIsoTemplate, rebuildIsoScriptName); err != nil {
+		return fmt.Errorf("creating the ISO rebuild script: %w", err)
+	}
+
+	cmd, rebuildLog, err := b.createIsoCommand(rebuildIsoLogFile, rebuildIsoScriptName)
 	if err != nil {
-		return fmt.Errorf("error running xorriso: %w", err)
+		return fmt.Errorf("preparing to build the new ISO: %w", err)
+	}
+	defer func() {
+		if err = rebuildLog.Close(); err != nil {
+			zap.L().Warn("failed to close ISO rebuild log file properly", zap.Error(err))
```

Comment:
The current format of the log file is text and not JSON so using structured logging looks weird in the output.
```suggestion
			zap.S().Warnf("Failed to close ISO rebuild log file properly: %s", err)
```

Reply:
Man, I really haven't looked at the zap APIs at all since it was set up. I didn't realize this distinction. I'll fix both mentioned instances.